bin/internal/prepare-workspace.sh

```diff
@@ -0,0 +1,121 @@
+#!/bin/sh
+
+################################################################################
+# This program and the accompanying materials are made available under the terms of the
+# Eclipse Public License v2.0 which accompanies this distribution, and is available at
+# https://www.eclipse.org/legal/epl-v20.html
+#
+# SPDX-License-Identifier: EPL-2.0
+#
+# Copyright IBM Corporation 2020
+################################################################################
+
+################################################################################
+# This script will run component `validate` and `configure` step if they are defined.
+#
+# This script take these parameters
+# - c:    INSTANCE_DIR
+# - t:    a list of component IDs separated by comma
+#
+# For example:
+# $ bin/internal/prepare-workspace.sh \
+#        -c "/path/to/my/zowe/instance" \
+#        -t "discovery,explorer-jes,jobs"
+################################################################################
+
+# if the user passes INSTANCE_DIR from command line parameter "-c"
+while getopts "c:t:" opt; do
+  case $opt in
+    c) INSTANCE_DIR=$OPTARG;;
+    t) LAUNCH_COMPONENTS=$OPTARG;;
+    \?)
+      echo "Invalid option: -$OPTARG" >&2
+      exit 1
+      ;;
+  esac
+done
+
+########################################################
+# prepare environment variables
+export ROOT_DIR=$(cd $(dirname $0)/../../;pwd)
+. ${ROOT_DIR}/bin/internal/prepare-environment.sh -c "${INSTANCE_DIR}"
+
+########################################################
+# Validate component properties if script exists
+ERRORS_FOUND=0
+for component_id in $(echo $LAUNCH_COMPONENTS | sed "s/,/ /g")
+do
+  component_dir=$(find_component_directory "${component_id}")
+  # backward compatible purpose, some may expect this variable to be component lifecycle directory
+  export LAUNCH_COMPONENT="${component_dir}/bin"
+  # FIXME: change here to read manifest `commands.validate` entry
+  VALIDATE_SCRIPT=${component_dir}/bin/validate.sh
```

Comment:
This is 100% my fault, but I've started trying to use lower_case variable names for non exported shell variables, so this might be an opportunity to convert more if you agree with that style guidance (which I think I got from the google style guide)

Reply:
Yes i do agree on this practice. Fixed.